frmts/vrt/vrtdataset.cpp

```diff
@@ -1481,6 +1481,58 @@ GDALDataset *VRTDataset::OpenVRTProtocol(const char *pszSpec)
                 argv.AddString("-eco");
             }
         }
+        else if (EQUAL(pszKey, "transpose"))
+        {
+
+            // FIXME: parse the argument from "array:ind1,ind2" and check the values
+            const CPLStringList aosTransposeTokens(
+                CSLTokenizeString2(pszValue, ":", 0));
+            if (aosTransposeTokens.size() != 2)
+            {
+                CPLError(CE_Failure, CPLE_IllegalArg,
+                         "Invalid transpose option: %s", pszValue);
+                return nullptr;
+            }
+            const CPLStringList aosTransposeIndex(
+                CSLTokenizeString2(aosTransposeTokens[1], ",", 0));
+            // fail if not two values
+            if (aosTransposeIndex.size() != 2)
+            {
+                CPLError(CE_Failure, CPLE_IllegalArg,
+                         "Invalid transpose option: %s", pszValue);
+                return nullptr;
+            }
+            int index_x = atoi(aosTransposeIndex[0]);
+            int index_y = atoi(aosTransposeIndex[1]);
+
+            auto poMDimDS = std::unique_ptr<GDALDataset>(
+                GDALDataset::Open(osFilename, GDAL_OF_MULTIDIM_RASTER));
```

Comment:
as we are reopening the source dataset, it means that "vrt:///some_dataset?some_option=some_value&transpose=...." would loose the effect of some_option.  So the evaluation of transpose must be done prior to this loop.

Reply:
ah, whoops, thanks I will re-work

Reply:
what option could be lost, though? They are all picked up and saved for Translate ( ? - I'm missing something about lost effects ??)

this works for example

```
gdalinfo "vrt://autotest/gdrivers/data/zarr/array_dimensions.zarr?a_srs=EPSG:3857&a_ullr=0,2,2,-2&transpose=/var:1,0"
```

I agree it's better that the transpose is isolated for good style and future-proofing, I just misunderstand the danger I think

Reply:
> what option could be lost, though?

"bands" for example: https://github.com/OSGeo/gdal/blob/16b13502e820ad07f47a668c797c57660a19b440/frmts/vrt/vrtdataset.cpp#L1198

Reply:
I force pushed a change to put the transpose loop on its own prior to sd_name/sd and added tests for exclusivity with those options. thanks!